src/utils/navigation.ts

```diff
@@ -237,6 +237,23 @@ export const getSurahInfoNavigationUrl = (chapterIdOrSlug: string): string =>
  */
 export const getReciterNavigationUrl = (reciterId: string): string => `/reciters/${reciterId}`;
 
+/**
+ * Get href link to the course page
+ *
+ * @param {string} courseSlug
+ * @returns {string} coursePageUrl
+ */
+export const getCourseNavigationUrl = (courseSlug: string): string => `/courses/${courseSlug}`;
+
+/**
+ * Get href link to the lesson page
+ *
+ * @param {string} courseSlug
+ * @returns {string} lessonPageUrl
+ */
+export const getLessonNavigationUrl = (courseSlug: string, lessonSlug: string): string =>
+  `/courses/${courseSlug}/lessons/${lessonSlug}`;
```

Comment:
nit: just a personal preference, but I prefer to remove `/lessons` so that the url is: `/courses/${courseSlug}/${lessonSlug}`

Reply:
I don't have a very strong opinion regarding this and each option has its pros and cons but in general I am leaning towards having `lessons` in the URL:

**Route 1: /courses/{courseSlug}/lessons/{lessonSlug}**

**Pros:**

* **Clearer hierarchy:** Explicitly specifies that the resource is a lesson within a course, aiding crawlers and users in understanding the URL structure.
* **Potential for semantic value:** If the course slug directly relates to the topic of the lesson, it can boost keyword targeting.
* **More flexible for future expansions:** Allows you to easily add additional subsections within lessons without changing the main URL structure.

**Cons:**

* **Longer URL:** May appear overwhelming to some users.

**Route 2: /courses/{courseSlug}/{lessonSlug}**

**Pros:**

* **Shorter and more concise:** Easier to remember and share.
* **Potentially cleaner aesthetic:** Looks less cluttered and can appear more user-friendly.

**Cons:**

* **Less explicit hierarchy:** The relationship between the course and lesson might be less apparent to crawlers, requiring additional internal linking or schema markup.
* **Less flexibility for future expansions:** Introducing sub-resources within lessons might require a URL revamp later.